### PR TITLE
Create config endpoint

### LINKED ISSUE #314: Create Configuration Endpoint
## Instructions
Create an endpoint for retrieving the configuration info of a running emulator.
This will be useful for integrations, such as the VSCode extension, to determine if an emulator can be connected to based on its service key or start up location.

### PR DESCRIPTION
**Pull request by DylanTinianov**
Closes #314 

## Description
Added /emulator/config endpoint to display configuration info for the ServiceKey

______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to GitHub issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-emulator/blob/master/CONTRIBUTING.md#styleguides)
- [ ] Updated relevant documentation
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer
- [ ] Added appropriate labels

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+)
+
+type ConfigInfo struct {
+	ServiceKey string `json:"service_key"`
+	StartupDir string `json:"startup_dir"`
```
Comment (sideninja) [inline server/config.go:30]: I don't think this is needed. If the service key is the same then it doesn't really matter where it was run.

File: server/config.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+var _ http.Handler = &ConfigInfo{}
+
+func NewConfigInfo(emulatorServer *EmulatorServer) *ConfigInfo {
+	startupDir, _ := os.Getwd()
+
+	return &ConfigInfo{
+		ServiceKey: fmt.Sprintf("0x%s", emulatorServer.blockchain.ServiceKey().Address.Hex()),
```
Comment (sideninja) [inline server/config.go:39]: This should return the public key which you can compare no the address. Also the value should better be `ServicePublicKey`

File: server/admin.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ import (
 const (
 	LivenessPath    = "/live"
 	MetricsPath     = "/metrics"
+	ConfigPath      = "/config"
```
Comment (sideninja) [inline server/admin.go:37]: This is better if it's under `/emulator` as that contains more admin endpoints for the emulator. So the route would be `/emulator/config`

**Review by sideninja — COMMENTED** [review]
Left comments

File: server/admin.go (hunk 3 above)
Comment (DylanTinianov) [reply server/admin.go]: Moved :)

File: server/config.go (hunk 2 above)
Comment (DylanTinianov) [reply server/config.go]: Actually we need to check the `ServicePrivateKey` which is  stored in flow.json

File: server/config.go (hunk 1 above)
Comment (DylanTinianov) [reply server/config.go]: Removed

File: server/emulator.go [hunk 4]
```diff
@@ -62,6 +62,9 @@ func NewEmulatorAPIServer(server *EmulatorServer, backend *backend.Backend, stor
 
 	router.HandleFunc("/emulator/storages/{address}", r.Storage)
 
+	config := NewConfigInfo(server)
+	router.Handle("/emulator/config", config)
```
Comment (bluesign) [inline server/emulator.go:66]: maybe just basic `HandleFunc` ? and a new function there, instead new file `config.go` ?

File: server/config.go [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+
+type ConfigInfo struct {
+	ServiceKey string `json:"service_key"`
+}
+
+var _ http.Handler = &ConfigInfo{}
+
+func NewConfigInfo(emulatorServer *EmulatorServer) *ConfigInfo {
+	return &ConfigInfo{
+		ServiceKey: emulatorServer.blockchain.ServiceKey().PrivateKey.String(),
```
[the remark this answers is not in the document]
Comment (sideninja) [reply server/config.go:34]: public yeah

File: server/emulator.go (hunk 4 above)
Comment (sideninja) [reply server/emulator.go:66]: +1

Comment (sideninja) [reply server/emulator.go:66]: I would add to admin.go as we have those gathered there

File: server/config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (DylanTinianov) [reply server/config.go:34]: The private key is what's exposed in the flow.json config, so we need to expose this to validate the emulator

[the remark this answers is not in the document]
Comment (sideninja) [reply server/config.go:34]: What he said. In vscode you will be able to derive public key from private key, I believe you can even use FCL for that.

[the remark this answers is not in the document]
Comment (sideninja) [reply server/config.go:34]: Although emulator is not meant to be used in production I can already see some weird case where someone will use it for some storage and then the private key will be leaked.

**Review by sideninja — APPROVED** [review]

File: server/config.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (DylanTinianov) [reply server/config.go]: Sounds good

**Pull request merged**